CONTRIBUTING.md

```diff
@@ -35,6 +37,18 @@ There are different ways to contribute to this project:
 3. Fork the repository and create pull requests
 4. Do testing
 
+### Investigate Before Contributing
+
+Before you start contributing and create issues and pull requests we ask you
+to browse through the existing issues and pull requests first to make sure that
+
+* the behaviour you would like to report or discuss, or
+* errors you would like to report or correct, or
+* enhancement you would like to request or propose, or
+* test-cases you would like to suggest
+
+are not covered by other issues and pull requests already.
+
```

Comment:
One last nitpick: This is really hard to read as a fluent sentence. I think it is sufficient if you tone down the list to the main words:

```
… first check whether an issue or pull request already exists for the error, enhancement, feature or test-case you have in mind. 
```

Also, if you instruct me to _make sure_ that the topic is not already discussed I would feel obliged to close all associated issues and PRs first :wink:

Reply:
Thanks for the suggestion. I used it to rephrase the sentence: commit f41f927aeb1091487c9200d3784eef946122ae91.